rfcs/2020-08-26-3191-host-metrics.md

```diff
@@ -0,0 +1,201 @@
+# RFC 3191 - 2020-08-26 - Collecting host-based metrics
+
+This RFC is to introduce a new metrics source to consume host-based metrics. The high level plan is to implement one (or more?) sources that collect CPU, disk, network, and memory metrics from Linux-based hosts.
+
+## Scope
+
+This RFC will cover:
+
+- A new source for host-based metrics, specifically:
+  - CPU
+  - Memory
+  - Disk
+  - Network
+- Collection on Linux, Windows, OSX.
+
+This RFC will not cover:
+
+- Other host metrics.
+- Other platforms
+
+## Motivation
+
+Users want to collect, transform, and forward metrics to better observe how their hosts are performing.
+
+## Internal Proposal
+
+Build a single source called `host_metrics` (name to be confirmed) to collect host/system level metrics.
+
+I've found a number of possible Rust-based solutions for implementing this collection, cross-platform.
+
+- https://crates.io/crates/heim (Rust)
+- https://lib.rs/crates/sys-info (Rust + C)
+- https://github.com/myfreeweb/systemstat (pure Rust)
+- https://docs.rs/sysinfo/0.3.19/sysinfo/index.html
+
+The Heim crate has a useful comparison doc showing the available tools and their capabilities.
+
+- https://github.com/heim-rs/heim/blob/master/COMPARISON.md
+
+For this implementation we're recommending the Heim crate based on platform and feature coverage.
+
+We'd use one of these to collect the following metrics:
+
+- `cpu_seconds_total` by mode (idle, nice, system, user) per CPU. (counter)
```

Comment:
@jszwedko given the Prometheus-style naming here. Does it make you want to adopt the same for the Apache metrics source for consistency? Just floating the question while we can.

Reply:
I could go either way on it. I think one difference here is there is no "provider name" for the metrics so we need to come up with one whereas with `apache_metrics` it is scraping already named metrics in Apache.

These is something nice about standardizing metric names in vector though. I agree that we should settle it before releasing the `apache_metrics` source though.

I'm wondering if @lukesteensen has any thoughts.

Reply:
Right. You can see the Prometheus `apache_exporter` [here](https://github.com/Lusitaniae/apache_exporter), notice how they follow the naming guidelines, which is what I'm proposing.

Reply:
Linking over to https://github.com/timberio/vector/pull/3519#issuecomment-681985323 for more discussion details specific to the apache metrics source.